bf2fc6cc711aee1a0c2a/kas-fleet-manager
# Add expires_at field to kafka request

## Description

Add a new read-only field called `expires_at` to the KafkaRequest schema which contains the calculation of the expiration time for instances that have a limit lifespan defined (in config/kafka-instance-types-configuration.yaml).
If a given instance size does not have a limit lifespan decided then expires_at is not set and thus not returned in the response.

Implications of this implementation:
* expires_at is derived from the creation time of the kafka instance and the lifespanSeconds from the size configuration. It is not stored in the DB
* If a kafka size has a lifespan set we ALWAYS set the expires_at
* If a kafka size does not have a lifespan we never set the expires_at
* expires_at can't be set by a user. If that is needed then we will change the behavior. The risk here is that the semantics of the expires_at field might change in that case which might be potentially breaking behavior
* If the lifespan of a kafka instance size is changed, the expiration time of already pre-existing instances will change too and thus the instances will be affected by the change. This happens in both scenarios of increasing the lifespan or decreasing it. This behavior was already happening before the implementation of this PR.

## Verification Steps

* Create a standard instance and verify that when getting that kafka instance through the API the expires_at field is not returned.
* Create a developer instance and verify that when getting that kafka instance through the API the expires_at is set at a time equal to created_at + lifespanSeconds. lifespanSeconds value should be the one set in the config/kafka-instance-types-configuration.yaml for the corresponding instance type-size.



## Checklist (Definition of Done)
